cli/cmd/run/run.go

```diff
@@ -28,9 +30,12 @@ func (r *Run) Run(ctx *clicontext.CLIContext) error {
 		isatty.IsTerminal(os.Stderr.Fd()) &&
 		isatty.IsTerminal(os.Stdin.Fd())
 
+	app, version := services.AppAndVersion(service)
+	svcName := fmt.Sprintf("%s@%s", app, version)
```

Comment:
I'm confused after so many changes, are we guaranteeing version now ? Would this never end up as `app@`

Reply:
Yeah this method will always return app and version. The bug in attach is that the parse logic doesn't take the real service name. it only takes app@version.